libcontainer/utils/utils_unix.go

```diff
@@ -364,3 +365,27 @@ func Openat(dir *os.File, path string, flags int, mode uint32) (*os.File, error)
 	}
 	return os.NewFile(uintptr(fd), dir.Name()+"/"+path), nil
 }
+
+// RetryOnEINTR takes a function that returns an error and calls it until it the error returned is
+// not EINTR.
+func RetryOnEINTR(fn func() error) error {
+	var err error
+	for {
+		err = fn()
+		if !errors.Is(err, unix.EINTR) {
+			break
+		}
+	}
+	return err
+}
```

Comment:
I (ever so slightly) would prefer something like this:

```go
for {
   err := fn()
      if !errors.Is(err, unix.EINTR) {
         return err
      }
   }
}
```

Reply:
Fixed, thanks!